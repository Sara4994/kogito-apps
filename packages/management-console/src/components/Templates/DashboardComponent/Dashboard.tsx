import {
  Nav,
  NavList,
  NavItem,
} from '@patternfly/react-core';
import React from 'react';
import { Redirect, Route, Link, Switch } from 'react-router-dom';
import { PageLayout } from '@kogito-apps/common/src/components';
import DataListContainer from '../DataListContainer/DataListContainer';
import ProcessDetailsPage from '../ProcessDetailsPage/ProcessDetailsPage';
import DomainExplorerDashboard from '../DomainExplorerDashboard/DomainExplorerDashboard';
import DomainExplorerLandingPage from '../DomainExplorerLandingPage/DomainExplorerLandingPage';
import ErrorComponent from '../../Molecules/ErrorComponent/ErrorComponent';
import NoDataComponent from '../../Molecules/NoDataComponent/NoDataComponent';
import './Dashboard.css';
import managementConsoleLogo from '../../../static/managementConsoleLogo.svg';

import { useGetQueryFieldsQuery } from '../../../graphql/types';

const Dashboard: React.FC<{}> = (props: any) => {
  const { pathname } = props.location;


  const PageNav = (
    <Nav aria-label="Nav" theme="dark">
      <NavList>
        <NavItem isActive={pathname === '/ProcessInstances'}>
          <Link to="/ProcessInstances">Process Instances</Link>
        </NavItem>
        <NavItem isActive={pathname === '/DomainExplorer'}>
          <Link to="/DomainExplorer">Domain Explorer</Link>
        </NavItem>
      </NavList>
    </Nav>
  );

  const BrandClick = () => {
    props.history.push('/ProcessInstances');
  };

  const getQuery = useGetQueryFieldsQuery();
  const availableDomains =
    !getQuery.loading && getQuery.data.__type.fields.slice(2);
  const domains = [];
  availableDomains && availableDomains.map(item => domains.push(item.name));
  return (

    <PageLayout PageNav={PageNav}
      BrandSrc={managementConsoleLogo}
      BrandAltText="Task Consol Logo"
      BrandClick={BrandClick}
    >
      <Switch>
        <Route
          exact
          path="/"
          render={() => <Redirect to="/ProcessInstances" />}
        />
        <Route exact path="/ProcessInstances" component={DataListContainer} />
        <Route
          exact
          path="/Process/:instanceID"
          component={ProcessDetailsPage}
        />
        <Route
          exact
          path="/DomainExplorer"
          component={DomainExplorerLandingPage}
        />
        <Route
          exact
          path="/DomainExplorer/:domainName"
          render={_props => (
            <DomainExplorerDashboard {..._props} domains={domains} />
          )}
        />
        <Route path="/NoData" component={NoDataComponent} />
        <Route path="*" component={ErrorComponent} />
      </Switch> {' '}
    </PageLayout>
  );
};

export default Dashboard;
