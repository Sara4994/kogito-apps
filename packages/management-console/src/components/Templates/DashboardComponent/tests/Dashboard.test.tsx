import React from 'react';
import { shallow } from 'enzyme';
import Dashboard from '../Dashboard';
import { MockedProvider } from '@apollo/react-testing';

const props: any = {
  location: {
    pathname: '/ProcessInstances'
  }
};

const mocks = [];

describe('Dashboard component tests', () => {
  it('snapshot testing', () => {
    const wrapper = shallow(
      <MockedProvider mocks={mocks}>
        <Dashboard {...props} />
      </MockedProvider>
    );
    expect(wrapper.find(Dashboard)).toMatchSnapshot();
  });
});
