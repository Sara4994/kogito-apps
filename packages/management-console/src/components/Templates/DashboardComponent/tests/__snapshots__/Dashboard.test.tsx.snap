// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component tests snapshot testing 1`] = `
<Dashboard
  location={
    Object {
      "pathname": "/ProcessInstances",
    }
  }
/>
`;
